src/AddUserScreen.tsx

```diff
@@ -12,9 +16,55 @@ export default function newUser() {
   const todayDate = new Date();
   const maxDate = todayDate.toISOString().split('T')[0];
 
-  function handleSubmit(e: any) {
+  const history = useHistory();
+  const [userSubmission, { loading }] = useMutation(CREATE_USER);
+
+  if (loading) {
+    return (
+      <div className='App'>
+        <header className='App-header'>
+          <ClipLoader />
+        </header>
+      </div>
+    );
+  }
+
+  const userMutation = async (
+    name: string,
+    email: string,
+    password: string,
+    phone: string,
+    birthDate: string,
+    role: string,
+  ) => {
```

Comment:
poderia passar um objeto com todos esses parametros, na função newUserValidation também seria bacana usar objeto

```suggestion
interface UserData {
    name: string,
    email: string,
    password: string,
    phone: string,
    birthDate: string,
    role: string,
}

  const userMutation = async (data: UserData) => {
```